mono/io-layer/wait.c

```diff
@@ -139,8 +139,10 @@ guint32 WaitForSingleObjectEx(gpointer handle, guint32 timeout,
 		goto done;
 	}
 	
-	if (timeout != INFINITE)
-		end = mono_100ns_ticks () + timeout * 1000 * 10;
+	if (timeout != INFINITE) {
+		wait_start = mono_100ns_ticks ();
+		timeout_in_ticks = (gint64)timeout * 10 * 1000; //can't overflow as timeout is 32bits
```

Comment:
It's ((guint64)timeout) * 10 * 1000.
I though regular associativity rules would be clear here.